stdlib/LinearAlgebra/src/LinearAlgebra.jl

```diff
@@ -388,6 +388,145 @@ const ⋅ = dot
 const × = cross
 export ⋅, ×
 
+"""
+    hadamard(a, b)
+    a ⊙ b
+
+For arrays `a` and `b`, perform elementwise multiplication.
+`a` and `b` must have identical `axes`.
+When either `a` or `b` is not an array, `hadamard(a, b)` defaults to `a*b`.
+
+`⊙` can be passed as an operator to higher-order functions.
+
+```jldoctest
+julia> a = [2, 3]; b = [5, 7];
+
+julia> a ⊙ b
+2-element Array{$Int,1}:
+ 10
+ 21
+
+julia> a ⊙ [5]
+ERROR: DimensionMismatch("Axes of `A` and `B` must match, got (Base.OneTo(2),) and (Base.OneTo(1),)")
+[...]
+```
+
+!!! compat "Julia 1.5"
+    This function requires at least Julia 1.5. In Julia 1.0-1.4 it is available from
+    the `Compat` package.
+"""
+function hadamard(A::AbstractArray, B::AbstractArray)
+    @noinline throw_dmm(axA, axB) = throw(DimensionMismatch("Axes of `A` and `B` must match, got $axA and $axB"))
+
+    axA, axB = axes(A), axes(B)
+    axA == axB || throw_dmm(axA, axB)
+    return map(hadamard, A, B)
+end
+hadamard(a, b) = a * b
```

Comment:
It would be great if others could chime in, whether this method should be restricted to only `Number`s. Since I believe that in mathematical literature scalar ⊙ matrix is usually not valid, I would lean towards doing the safer thing by throwing an error if the arguments aren't either both numbers or both arrays. Otherwise, just plain `.*` should do the trick in most situations anyways.